arch/lkl/include/asm/thread_info.h

```diff
@@ -8,6 +8,15 @@
 #include <asm/processor.h>
 #include <asm/host_ops.h>
 
+#include <linux/signal_types.h>
+#include <linux/spinlock_types.h>
+
+struct ksignal_list_node
+{
+	struct ksignal sig;
+	volatile struct ksignal_list_node *next;	/* consider using the kernel lists, but they are doubly linked and clumsy in this simple case */
```

Comment:
Doesn't Linux have `SLIST`?

Why is this `volatile`?

Reply:
More left overs. All this code had been working and I broke it merging. Prior to finding why I tried a bunch of 'it can't be but...' things.

Linux lists seem to be doubly linked with a complicated API. I just looking at it.